prototype/core/geo-json/src/main/java/org/eclipse/sensinact/gateway/geojson/Coordinates.java

```diff
@@ -36,4 +38,45 @@ public class Coordinates {
      * The elevation will be {@link Double#NaN} if not set
      */
     public double elevation = Double.NaN;
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(longitude, latitude, elevation);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof Coordinates) {
```

Comment:
This check isn't considered best practice for `equals` unless the type is `final`. If someone subclasses Coordinates then this test will be true, even though we're doing a potentially incomplete check.

```suggestion
        if (obj != null && obj.getClass() == Coordinates.class) {
```